lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+    
+    slider.appendChild(loadingPlace).appendChild(loadingImage);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderElements(sliderData);
+        autoSwitchSlider(sliderData);
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderElements = (sliderData) => {
+    renderSliderItems(sliderData);
+    getPageNumber(sliderData);
+    renderButtons(sliderData);
+    renderDotPagination(sliderData);
+    isActiveFirstItem();
+}
+
+let autoPlay;
+const autoSwitchSlider = (sliderData) =>{
+    autoPlay = setInterval(() =>{
+        currentIndex ++;
+        if(currentIndex >= sliderData.length){
+            currentIndex = 0;
+        }
+        switchSlider(sliderData,currentIndex);
+    }, 3000);
+}
+
+const resetAutoPlay = (sliderData) => {
+    clearInterval(autoPlay);
+    autoSwitchSlider(sliderData);
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const image of sliderData){
+        const sliderItem = createAttributedElements({tag:"div",attrObj:{class:"slider__item"}});
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+
+    slider.appendChild(fragment);
+}
+
+const getPageNumber = (sliderData) => {
+    const pageNumber    = createAttributedElements({tag:"p",attrObj:{class:"page-number"},str:`/${sliderData.length}`});
+    const currentNumber = createAttributedElements({tag:"span",attrObj:{id:"js-currentNumber",class:"current-number"},str:"1"});
+
+    slider.appendChild(pageNumber).insertAdjacentElement("afterbegin",currentNumber);
+}
+
+const renderButtons = (sliderData) => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn",disabled:true}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+    addClickEventForButton(sliderData);
+}
+
+const renderDotPagination = (sliderData) => {
```

Comment:
IMO
The name ‘Dot' does not seem appropriate here. It's a name that depends on the design.
[reference](https://github.com/haru-programming/JavaScript-lessons/pull/22#discussion_r841141656)

Reply:
Exactly.
Thank you for teaching me!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/20/commits/ab937a229a86c4d5dab13e1b1ba6f06f95d8f0f6